dlls/kernelx/kernelx.cpp

```diff
@@ -337,17 +337,121 @@ LPVOID VirtualAllocEx_X(
 
     return ret;
 }
+const char* AllocationTypeToString(DWORD allocType)
+{
+    switch (allocType)
+    {
+    case MEM_COMMIT: return "MEM_COMMIT";
+    case MEM_RESERVE: return "MEM_RESERVE";
+    case MEM_RESET: return "MEM_RESET";
+    case MEM_RESET_UNDO: return "MEM_RESET_UNDO";
+    case MEM_LARGE_PAGES: return "MEM_LARGE_PAGES";
+    case MEM_PHYSICAL: return "MEM_PHYSICAL";
+    case MEM_TOP_DOWN: return "MEM_TOP_DOWN";
+    case MEM_WRITE_WATCH: return "MEM_WRITE_WATCH";
+    default: return "UNKNOWN_ALLOCATION_TYPE";
+    }
+}
+
+std::string ProtectionToString(DWORD protect)
+{
+    std::string result;
+
+    switch (protect & 0xFF) // Base protection (lower byte)
+    {
+    case PAGE_NOACCESS: result += "PAGE_NOACCESS"; break;
+    case PAGE_READONLY: result += "PAGE_READONLY"; break;
+    case PAGE_READWRITE: result += "PAGE_READWRITE"; break;
+    case PAGE_WRITECOPY: result += "PAGE_WRITECOPY"; break;
+    case PAGE_EXECUTE: result += "PAGE_EXECUTE"; break;
+    case PAGE_EXECUTE_READ: result += "PAGE_EXECUTE_READ"; break;
+    case PAGE_EXECUTE_READWRITE: result += "PAGE_EXECUTE_READWRITE"; break;
+    case PAGE_EXECUTE_WRITECOPY: result += "PAGE_EXECUTE_WRITECOPY"; break;
+    default: result += "UNKNOWN_BASE_PROTECTION"; break;
+    }
+
+    // Extra modifiers (higher bits)
+    if (protect & PAGE_GUARD)
+        result += " | PAGE_GUARD";
+    if (protect & PAGE_NOCACHE)
+        result += " | PAGE_NOCACHE";
+    if (protect & PAGE_WRITECOMBINE)
+        result += " | PAGE_WRITECOMBINE";
+
+    return result;
+}
 
 LPVOID VirtualAlloc_X(
     LPVOID lpAddress,
     SIZE_T dwSize,
-    DWORD  flAllocationType,
-    DWORD  flProtect
+    DWORD flAllocationType,
+    DWORD flProtect
 )
 {
-    return VirtualAllocEx_X(GetCurrentProcess(), lpAddress, dwSize, flAllocationType, flProtect);
+    printf("[VirtualAlloc_X] Request:\n");

```

Comment:
Same issue as logging successful allocations, these logs will be extremely noisy and will harm performance. If you really want to log this information, it should probably be limited to debug builds at the very least.